tensorly/metrics/similarity.py

```diff
@@ -0,0 +1,83 @@
+from typing import List
+
+import numpy as np
+
+import tensorly as tl
+
+# Authors: Hratch Baghdassarian <hmbaghdassarian@gmail.com>, Erick Armingol <earmingol14@gmail.com>
+# similarity metrics for tensor decompsoitions
+
+def stack_loading_matrices(factors: List[np.ndarray]) -> np.ndarray:
+    """Vertically stack the loading/factor matrices.
+
+    Parameters
+    ----------
+    factors : List[np.ndarray]
+        The loading/factor matrices [A^1 ... A^R] for a decomposed low-rank tensor from its factors.
+
+    Returns
+    -------
+    np.ndarray
+        A vertically stacked matrix of shape [sum(tensor.shape)xR)] 
+    """
+
+    if len({A.shape[1] for A in factors}) != 1:
+        raise ValueError('factors should be a list of loading matrices')
+    return tl.concatenate(factors)
+
+def calc_corrindex(C: np.ndarray) -> float:
+    """Calculate the CorrIndex.
+
+    Parameters
+    ----------
+    C : np.ndarray
+        absolute value of the product of two MxN matrices that have had their columns L2 normalized, 
+        taking the complex conjugate transpose of the first matrix -- C = |X1^H * X2|
+
+    Returns
+    -------
+    float
+        CorrIndex metric [0,1]; lower score means higher between matrices
+    """
+    if C.shape[1] != C.shape[0]:
+        raise ValueError('C must be a square matrix')
+
+    MN = C.shape[1]*C.shape[0] #TODO: N notation in equation 8 unclear, make sure this is correct
```

Comment:
what about size or total size? Also better to use tl.shape(C)[1] just to make it backend agnostic.